homeassistant/components/alarm_control_panel/significant_change.py

```diff
@@ -26,13 +26,16 @@ def async_check_significant_change(
     if old_state != new_state:
         return True
 
-    old_attrs_s = set(old_attrs.items())
-    new_attrs_s = set(new_attrs.items())
+    old_attrs_s = set(
+        {k: v for k, v in old_attrs.items() if k in SIGNIFICANT_ATTRIBUTES}.items()
+    )
+    new_attrs_s = set(
+        {k: v for k, v in new_attrs.items() if k in SIGNIFICANT_ATTRIBUTES}.items()
+    )
     changed_attrs: set[str] = {item[0] for item in old_attrs_s ^ new_attrs_s}
```

Comment:
maybe this logic could be moved in a helper method under `homeassistant.helpers.significant_change` 🤔